Move uniform_() to DistributionTemplates(Migrate CPU uniform_ from TH to ATen)

pbelevich opened the pull request:
`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35585 rand() and uniform_() for complex dtype
* **#35580 Move uniform_() to DistributionTemplates(Migrate CPU uniform_ from TH to ATen)**

Differential Revision: [D20820221](https://our.internmc.facebook.com/intern/diff/D20820221)

pbelevich committed:
Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)

[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954




[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954


Differential Revision: [D20820221](https://our.internmc.facebook.com/intern/diff/D20820221)

[ghstack-poisoned]

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954


Differential Revision: [D20820221](https://our.internmc.facebook.com/intern/diff/D20820221)

[ghstack-poisoned]

anjali411 commented on aten/src/ATen/native/DistributionTemplates.h:
```diff
@@ -262,4 +260,29 @@ Tensor normal_impl(const Tensor& mean, const Tensor& std, Generator gen) {
+
+template<template<typename> class uniform_kernel, typename RNG>
+at::Tensor& uniform_impl_(at::Tensor& self, double from, double to, at::Generator generator) {
+  AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, self.scalar_type(), "check_uniform_bounds", [&] {
```
why does the complex handling not show up here? I know you made the changes here: https://github.com/pytorch/pytorch/pull/35585

pbelevich replied:
why should it be here?

anjali411 commented on aten/src/ATen/test/cpu_rng_test.cpp:
```diff
@@ -96,13 +102,15 @@ class RNGTest : public ::testing::Test {
       .impl("aten::normal.float_Tensor",      torch::dispatch(DispatchKey::CustomRNGKeyId, CppFunction::makeUnboxedOnly(normal_float_Tensor)))
       .impl("aten::normal.Tensor_Tensor",     torch::dispatch(DispatchKey::CustomRNGKeyId, CppFunction::makeUnboxedOnly(normal_Tensor_Tensor)))
+      // Uniform
+      .impl("aten::uniform_",     torch::dispatch(DispatchKey::CustomRNGKeyId, CppFunction::makeUnboxedOnly(uniform_)))
```
what does this line do?

pbelevich replied:
it registers custom `uniform_ ` with dispatch key `CustomRNGKeyId ` to "aten::uniform_" schema

anjali411 reviewed (approved):
LGTM :)

pbelevich committed:
Update on "Move uniform_() to DistributionTemplates(Migrate CPU uniform_ from TH to ATen)"

`uniform_kernel_cpu` is based on https://github.com/pytorch/pytorch/pull/30954


Differential Revision: [D20820221](https://our.internmc.facebook.com/intern/diff/D20820221)

[ghstack-poisoned]
